packages/jest-runtime/src/index.ts

```diff
@@ -553,6 +553,69 @@ export default class Runtime {
       return globals;
     }
 
+    if (specifier.startsWith('data:')) {
+      if (
+        this._shouldMock(
+          referencingIdentifier,
+          specifier,
+          this._explicitShouldMockModule,
+          {conditions: this.esmConditions},
+        )
+      ) {
+        return this.importMock(referencingIdentifier, specifier, context);
+      }
+
+      const fromCache = this._esmoduleRegistry.get(specifier);
+
+      if (fromCache) {
+        return fromCache;
+      }
+
+      const match = specifier.match(
+        /^data:text\/javascript;(charset=utf-8|base64),(.*)$/,
```

Comment:
can you use named capture groups? instead of indexes. II find it more readable 🙂